azurerm/resource_arm_bot_channel_slack.go

```diff
@@ -0,0 +1,219 @@
+func resourceArmBotChannelSlackCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).bot.ChannelClient
+	ctx := meta.(*ArmClient).StopContext
+
+	resourceGroup := d.Get("resource_group_name").(string)
+	botName := d.Get("bot_name").(string)
+
+	if features.ShouldResourcesBeImported() && d.IsNewResource() {
+		existing, err := client.Get(ctx, resourceGroup, string(botservice.ChannelNameSlackChannel), botName)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("Error checking for presence of creating Bot Channel Slack (Resource Group %q / Bot %q): %+v", resourceGroup, botName, err)
+			}
+		}
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_bot_channel_slack", *existing.ID)
+		}
+	}
+
+	channel := botservice.BotChannel{
+		Properties: botservice.SlackChannel{
+			Properties: &botservice.SlackChannelProperties{
+				ClientID:                utils.String(d.Get("client_id").(string)),
+				ClientSecret:            utils.String(d.Get("client_secret").(string)),
+				VerificationToken:       utils.String(d.Get("verification_token").(string)),
+				LandingPageURL:          utils.String(d.Get("landing_page_url").(string)),
+				IsEnabled:               utils.Bool(true),
+				RegisterBeforeOAuthFlow: utils.Bool(true),
+			},
+			ChannelName: botservice.ChannelNameSlackChannel1,
+		},
+		Location: utils.String(d.Get("location").(string)),
+		Kind:     botservice.KindBot,
+	}
+
+	if _, err := client.Create(ctx, resourceGroup, botName, botservice.ChannelNameSlackChannel, channel); err != nil {
+		return fmt.Errorf("Error issuing create request for Bot Channel Slack (Resource Group %q / Bot %q): %+v", resourceGroup, botName, err)
+	}
+
+	resp, err := client.Get(ctx, resourceGroup, botName, string(botservice.ChannelNameSlackChannel))
+	if err != nil {
+		return fmt.Errorf("Error making get request for Bot Channel Slack (Resource Group %q / Bot %q): %+v", resourceGroup, botName, err)
+	}
+
+	if resp.ID == nil {
+		return fmt.Errorf("Cannot read Bot Channel Slack (Resource Group %q / Bot %q): %+v", resourceGroup, botName, err)
+	}
+
+	d.SetId(*resp.ID)
+
+	return resourceArmBotChannelSlackRead(d, meta)
+}
+
+func resourceArmBotChannelSlackRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).bot.ChannelClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	botName := id.Path["botServices"]
+	resp, err := client.Get(ctx, id.ResourceGroup, botName, string(botservice.ChannelNameSlackChannel))
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] Error reading Bot Channel Slack (Resource Group %q / Bot %q): %+v", id.ResourceGroup, botName, err)
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("Error reading Bot Channel Slack (Resource Group %q / Bot %q): %+v", id.ResourceGroup, botName, err)
```

Comment:
can we match the ordering with other resources:

```suggestion
		return fmt.Errorf("Error reading Channel Slack for Bot %q (Resource Group %q): %+v", botName, id.ResourceGroup, err)
```